src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
 export interface Figure {
+  shape: string;
+  color: string;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle side length is not correct');
+    }
+
+    if ((a + b <= c) || (a + c <= b) || (b + c <= a)) {
+      throw new Error('Not a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPerimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number
+      = Math.sqrt(
+        halfPerimeter
+        * (halfPerimeter - this.a)
+        * (halfPerimeter - this.b)
+        * (halfPerimeter - this.c),
+      );
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius length is not correct');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?

Reply:
Maybe you're right)))